plat/ti/k3/common/bl31.c

```diff
@@ -33,6 +58,30 @@ void bl31_early_platform_setup(bl31_params_t *from_bl2,
 	/* There are no parameters from BL2 if BL31 is a reset vector */
 	assert(from_bl2 == NULL);
 	assert(plat_params_from_bl2 == NULL);
+
+	/* Populate entry point information for BL32 */
```

Comment:
Does your platform always has secure image to be booted alone side non secure image ?
Should this be guarded by PRELOADED_BL32_BASE #ifdef?

Reply:
If it does not have a secure image, then ATF should be compiled with "SPD=none". The #ifdef guard would do nothing in this case as the board.mk file will populate PRELOADED_BL32_BASE with a default value if it is empty.

Reply:
My point was, what if there is no secure image.
 is this code still executes then?

Reply:
No, if you do not have a secure image and compile without SPD=something then ATF never fetches this address so it will not be used.